webapp/src/shared/components/Game/Game.tsx

```diff
@@ -1,78 +1,157 @@
 import * as React from 'react';
-import { CANVAS_WIDTH, CANVAS_HEIGHT } from 'pong-engine';
 import {
-  useGameControls,
-  useGameAnimation,
-  useGameEngine,
-  useOnlineGame,
-} from './hooks';
-import { GameStateContextProvider } from './context';
-import {
-  StateMachineContextProvider,
-  useStateMachineContext,
-} from './state-machine/context';
-import Button, { ButtonVariant } from '../Button/Button';
+  CANVAS_WIDTH,
+  CANVAS_HEIGHT,
+  GameState,
+  GameCommand,
+} from 'pong-engine';
+import { useGameControls, useGameAnimation } from './hooks';
+import { GameStateContextProvider } from './context/gameStateContext';
 import Text, { TextVariant } from '../Text/Text';
 import GameSpinner from '../GameSpinner/GameSpinner';
 import Header from '../Header/Header';
 import { IconVariant } from '../Icon/Icon';
 import { useNavigation } from '../../hooks/UseNavigation';
 
 import './Game.css';
+import socket from '../../socket';
+import { GameInfo, WsException } from '../../types';
+import { useAuth } from '../../hooks/UseAuth';
+import { useNotificationContext } from '../../context/NotificationContext';
+import { useNavigate } from 'react-router-dom';
+import { PLAY_URL } from '../../urls';
+import Button, { ButtonVariant } from '../Button/Button';
+
+const GAME_COMMAND = 'gameCommand';
+const UPDATE_GAME = 'updateGame';
+const JOIN_GAME = 'joinGame';
+const LEAVE_GAME = 'leaveGame';
+const GAME_JOINED = 'gameJoined';
+const GAME_NOT_FOUND = 'gameNotFound';
+const GAME_FINISHED = 'gameFinished';
 
-const Play = () => {
-  const { renderFrame } = useGameAnimation();
+type GameProps = {
+  gameId: string;
+};
+
+const Play = ({ gameId }: GameProps) => {
+  const { notify, warn } = useNotificationContext();
+  const { renderMultiplayerFrame } = useGameAnimation();
   const canvasRef = React.useRef<HTMLCanvasElement>(null);
-  const requestFrameRef = React.useRef<number | null>(null);
   const [score, setScore] = React.useState<number>(0);
-  const { sendGameCommand, updateGame } = useOnlineGame();
-  // Refactor para decorar o hacer HOC con estos componentes
-  // De esta forma usar versión online de los controles y versión online del gameEngine
-  // usamos hook online dentro de la versión decorada
+  const [opponentScore, setOpponentScore] = React.useState<number>(0);
+  const isPlayerOneRef = React.useRef(true);
+  const isPlayerRef = React.useRef(false);
+  const gameStateRef = React.useRef<GameState | null>(null);
+  const { authUser } = useAuth();
+  const navigate = useNavigate();
+  const [gameJoined, setGameJoined] = React.useState(false);
+  const sendGameCommand = React.useCallback(
+    (command: GameCommand) => {
+      if (isPlayerRef.current && gameJoined) {
+        socket.emit(GAME_COMMAND, {
+          command,
+          gameRoomId: gameId,
+        });
+      }
+    },
+    [gameId, gameJoined],
+  );
```

Comment:
Why did you delete the online hook and move the socket handling here? I think we are mixing things up but I'm sure you have a good reason :D

Reply:
Updated. I've added it back and applied the changes to the current implementation.